api/tests/routes/backoffice/admin_test.py

```diff
@@ -103,6 +103,55 @@ def test_update_role_with_empty_permissions(self, authenticated_client):
         db.session.refresh(role)
         assert role.permissions == []
 
+    def test_log_update_role(self, authenticated_client):
+        perms = perm_factories.PermissionFactory.create_batch(4)
+        old_perms = [perms[0], perms[1]]
+        role_to_edit = perm_factories.RoleFactory(permissions=old_perms)
+        role_not_to_edit = perm_factories.RoleFactory(permissions=old_perms)
+
+        new_perms = [perms[1], perms[3]]
+        base_form = {}
+        for perm in perms:
+            base_form[perm.name] = perm in new_perms
+
+        response = self.post_to_endpoint(authenticated_client, role_id=role_to_edit.id, form=base_form)
+        assert response.status_code == 303
+
+        action = history_models.ActionHistory.query.one()
+        assert action.actionType == history_models.ActionType.ROLE_PERMISSIONS_CHANGED
+        assert action.authorUserId is not None
+        assert action.extraData == {
+            "role_name": role_to_edit.name,
+            "modified_info": {
+                role_not_to_edit.permissions[0].name: {"new_value": False, "old_value": True},
+                role_to_edit.permissions[1].name: {"new_value": True, "old_value": False},
+            },
+        }
+
+    class LogHistoryTest(GetEndpointHelper):
```

Comment:
Problème d'indentation, ta classe se retrouve dans `UpdateRoleTest`, mais c'est un autre endpoint.
Aussi, peux-tu la renommer plus clairement en fonction de l'endpoint ?
```suggestion
class GetRolesHistoryTest(GetEndpointHelper):
```